netket/sampler/rules/hamiltonian.py

```diff
@@ -47,26 +47,36 @@ class HamiltonianRule(MetropolisRule):
     :class:`netket.sampler.MetropolisSamplerNumpy`.
 
     Attributes:
-        Ô: The (hermitian) operator giving the transition amplitudes.
+        Ô: The (hermitian) operator giving the transition amplitudes.
 
     """
 
-    Ô: AbstractOperator = struct.field(pytree_node=False)
+    Ô: AbstractOperator = struct.field(pytree_node=False)
```

Comment:
The variable name changed from Ô to Ô.  Exciting!

On a serious note, it actually did change from `'O\xcc\x82'` to `'\xc3\x94'` (which, after normalization, is the same character). _I think_ this does not break any code using this variable, as Python normalizes identifiers. It would be nice, though, to have a way to avoid these spurious diffs. I wonder if we could automate this - I don't think `black` can do that, unfortunately.

Reply:
Yeah, I also noticed because apparently Gitub does not render correctly the Zero-width Joiner \xcc ...

I think that's because I edited in sublime-text instead of vscode that I had used when writing that code, and their tab completion apparently yield different results...

I have no idea

Reply:
I know unicode identifiers look pretty, but maybe we should just write `Ohat`? Or, actually, `.operator` or `.hamiltonian` (the class is called `HamiltonianRule`, after all). At least in the public interface, that saves some headache for us (see these diffs) and users whose tab completion might not work well with these.

Reply:
about the class, yeah, I would like to rename it to `OperatorRule` and `MetropolisOperator` but @gcarleo had some reservations I don't remember about that.

This is really an operator, not an hamiltonian.

(About unicode: I would like to gripe. For ### sake! it's 2021! why can't python editors have decent support or what makes formulas easier to read?)

Reply:
> (About unicode: I would like to gripe. For ### sake! it's 2021! why can't python editors have decent support or what makes formulas easier to read?)

In general, I concur with this point.

In this specific instance, however, `rule.O` (or `rule.op` or similar) seems to me to be as readable as `rule.Ô` (in fact, the Ô looks very close to O in the monospace fonts I have here and its hard to see whether there is a tilde, a hat, or something else on top of the O). `self.op` is easier to type even when you have editor support for fancy symbols.